Use shared `SummarySection` in Programs Report page.

tomalec opened the pull request:
_This PR is based on #569 to reduce the conflict with shared summary section_

### Changes proposed in this Pull Request:

Use shared `SummarySection` on the Programs Report page.
Update mocked metrics data to match the API schema.
Update mocked programs data to use correct metric names.

Implements part of #242 

This is a baby step to make the programs report page follow products one.


### Screenshots:

All programs / `missingFreeListingsData=true`
![Summary section on Programs Report Page for all programs](https://user-images.githubusercontent.com/17435/117726615-19012800-b1e7-11eb-93ce-da5e2e69dd76.gif)

Free listings only / `missingFreeListingsData=na`
![Summary section on Programs Report Page for Free Listings only](https://user-images.githubusercontent.com/17435/117726916-94fb7000-b1e7-11eb-9913-adaaf59979d1.gif)


Paid campaigns only / `missingFreeListingsData`
![Summary section on Programs Report Page for Paid Campaigns only](https://user-images.githubusercontent.com/17435/117726905-90cf5280-b1e7-11eb-8d1b-af4e58a36eca.gif)




### Detailed test instructions:

1. Open Programs report page https://gla1.test/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Freports%2Fprograms
2. The currency format displayed on the summary section should align with store's currency setting
3. The default selected summary metric should be the first metric
4. Click on summary metrics, the selected metric and table's order column should be changed respectively.

:warning: Metrics are still using mocked data, so switching program filter will make no effect. However, to test different scenarios you can use `missingFreeListingsData` URL param:

1. [none](https://gla1.test/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Freports%2Fprograms) - when only paid programs are filtered (or we are somewhere in a far future when free listings also have all metrics) 
2. [`missingFreeListingsData=na`](https://gla1.test/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Freports%2Fprograms&orderby=totalSpend&missingFreeListingsData=na) - when only free listings are chosen
3. [`missingFreeListingsData=true`](https://gla1.test/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Freports%2Fprograms&orderby=totalSpend&missingFreeListingsData=true) - when free listings and paid campaigns are chosen together.

tomalec committed:
Use shared SummarySection in Programs Report page.

Update mocked metrics data to match the API schema.
Update mocked programs data to use correct metric names.

eason9487 reviewed (approved):
Testing well and LGTM. Left a memo comment for updating related props changes which were made after this PR was opened.

tomalec committed:
Merge branch 'feature/244-connect-data' into feature/242-summary-section

tomalec committed:
Use latest API of SummarySection in ProgramsReport.

Addresses https://github.com/woocommerce/google-listings-and-ads/pull/576#discussion_r629839589
